lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback, args) {
```

Comment:
I don't buy it that `new Immediate` was slower than returning static object. It should be as fast while providing better debug traces (because it is named) - both get the same hidden class exactly. 

Putting object properties on the prototype is a perf killer. Not suggesting we stick to that. Definitely going to see an improvement compared to prototype properties anyway.